pageserver/src/tenant.rs

```diff
@@ -1700,6 +1710,11 @@ impl Tenant {
                 .await?;
         }
 
+        // Perhaps we did no work and the walredo process has been idle for some time:
+        // give it a chance to shut down to avoid leaving walredo process running indefinitely.
+        self.walredo_mgr
+            .maybe_quiesce(self.get_compaction_period() * 10);
```

Comment:
If this has to be the implementation, I would rather have this in the tasks file, since perhaps it is less clear that compaction only happens via a task in prod, never through any http api.

Reply:
a2f352907

Reply:
I was wrong; we have no http api which would call Tenant::compaction_iteration. Keeping it moved to consider killing walredo regardless of compaction errors.